tripy/examples/nanogpt/model.py

```diff
@@ -137,16 +132,15 @@ def __init__(self, config):
         super().__init__()
         self.wte = tp.Embedding(config.vocab_size, config.embedding_size, dtype=config.dtype)
         self.wpe = tp.Embedding(config.block_size, config.embedding_size, dtype=config.dtype)
-        self.h = [Block(config) for _ in range(config.num_layers)]
+        self.h = tp.Sequential(*[Block(config) for _ in range(config.num_layers)])
         self.ln_f = tp.LayerNorm(config.embedding_size)
-        self.pos = tp.reshape(tp.arange(0, config.seq_len, dtype=tp.int32), (1, config.seq_len))
 
     def __call__(self, idx):
         tok_emb = self.wte(idx)  # token embeddings of shape (batch_size, seq_len, embedding_size)
-        pos_emb = self.wpe(self.pos[:, : idx.shape[1]])  # position embeddings of shape (seq_len, embedding_size)
+        pos = tp.unsqueeze(tp.arange(idx.shape[1], dtype=tp.int32), 0)
```

Comment:
I think this is incorrect, `config.seq_len` will be the max seq len and `idx.shape[1]` will be the current sequence length.

Reply:
Ah so we need to constrain it to the `config.seq_len`?

Reply:
Updated.